### ISSUE TITLE
Granular minification

### ISSUE THREAD

**Issue by guybedford** [systemjs/builder#209]
We could apply minification individually to each load record, which would allow the compile cache to cached minified sources.

There is no minification loss of individually minifying loads, as they have their own scopes.

This way the final concatOutput step only needs to concatenate source maps, and not run any minification process at all.

/cc @ohjames

**Comment by ohjames** [comment]
I wonder if this would be needed: currently the cache API isn't used internally and I suppose it might not ever be given the command-line tool is always a one-shot process.

Consumers of the cache API are likely going to be build systems and will probably turn off builder's own minification in favour of their own minification plugins.

**Comment by guybedford** [comment]
Roughly 50% of compile time is spent minifying, so that being able to cache minifications per load would be a major benefit.

**Comment by ohjames** [comment]
Would you save the cache out to disk then so it could be re-used between runs of the builder command or something?

In the case of build systems consuming the `builder` as an API, they'll be doing their incremental rebuild stuff without minification right?

**Comment by ohjames** [comment]
At least for me I've never cared about incremental rebuilds of production code, I always consider minified builds a rare one-shot process.

**Comment by guybedford** [comment]
I see what you mean - production builds and development server builds are two separate use cases. But speeding up production builds with incremental disk-caching including minfication is still beneficial. Eg we could use a temporary folder to store the load caches in jspm to make builds much quicker.

**Comment by guybedford** [comment]
Which reminds me - we should probably include the load metadata and transpilation options in the hash of the load cache.

**Comment by ohjames** [comment]
Yeah it would make much more sense to start caching minified output if you can reuse the cache over multiple build instances. It might be good to store both the minified and non-minified translated sources in the cache so that two consumers 

About hashing the options: if do two builds with different options alternately sequenced, they will keep destroying the other's cache, you might wanna somehow encode the options as part of the hash key (which would be easy with an ES6 Map, without it would be doable but less efficient).
